pkg/disk/unmarshal.go

```diff
@@ -22,3 +24,35 @@ func unmarshalYAMLviaJSON(u json.Unmarshaler, unmarshal func(any) error) error {
 	}
 	return nil
 }
+
+func unmarshalJSONPayload(data []byte) (PayloadEntity, error) {
+	var payload struct {
+		Payload     json.RawMessage `json:"payload"`
+		PayloadType string          `json:"payload_type,omitempty"`
+	}
+	if err := json.Unmarshal(data, &payload); err != nil {
+		return nil, fmt.Errorf("cannot peek payload: %w", err)
+	}
+	if payload.PayloadType == "" {
+		if len(payload.Payload) > 0 {
+			return nil, fmt.Errorf("can build payload: empty payload type but payload is: %q", payload.Payload)
```

Comment:
`can build payload`?
I assume that's meant to be `cannot`.

Reply:
Uh, silly me, thank you! Indeed, "cannot" - fixed now.